app/controllers/loyalty_program_data_controller.rb

```diff
@@ -42,6 +59,14 @@ def create
   def update
     respond_to do |format|
       if @loyalty_program_datum.update(loyalty_program_datum_params)
+
+
+        # I want to add points to a particular loyalty program data with loyalty program id = params[:id], but dont know how to do
+        # @loyalty_program_datum.where(loyalty_program_id: params[:id]).points += loyalty_program_datum_params[:points].to_d
+
+ 
+        @transaction = Transaction.create(amount: loyalty_program_datum_params[:points].to_d, loyalty_program_data_id: params[:id], status: 0, account_id: current_user.id)
+        Thread.new(AccrualProcessor.convert_to_accrual(@transaction)) 
```

Comment:
why do you need another thread for this? have you tested this out? seems like this could break

A simpler way would be the use of functions